### PR TITLE
add a note on checking for equality with singletons

### PR DESCRIPTION
**Pull request by KristofferC**
Ref https://github.com/JuliaLang/julia/issues/28695

### CODE REVIEW

File: doc/src/manual/performance-tips.md [hunk 1]
```diff
@@ -1541,3 +1541,11 @@ will not require this degree of programmer annotation to attain performance.
 In the mean time, some user-contributed packages like
 [FastClosures](https://github.com/c42f/FastClosures.jl) automate the
 insertion of `let` statements as in `abmult3`.
+
+# Checking for equality with a singleton
+
+When checking if a value is equal to some singleton it can be be
```
Comment (ararslan) [inline doc/src/manual/performance-tips.md]: > be be

File: doc/src/manual/performance-tips.md [hunk 2]
```diff
@@ -1541,3 +1541,11 @@ will not require this degree of programmer annotation to attain performance.
...
+better for performance to check for identicality (`===`) instead of
+equality (`==`). The same advice applies to using `!==` over `!=`.
+These type of checks frequently occurs e.g. when implementing the iteration
```
Comment (ararslan) [inline doc/src/manual/performance-tips.md]: Should be "occur" rather than "occurs"

**Comment by ararslan** [comment]
LGTM other than the noted comments

**Comment by alyst** [comment]
Sounds like a black magic ;) Is it possible to include some brief explanation for the performance improvement?

**Pull request merged**